apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -231,6 +231,10 @@ type BMCDetails struct {
 	// insecure because it allows a man-in-the-middle to intercept the
 	// connection.
 	DisableCertificateVerification bool `json:"disableCertificateVerification,omitempty"`
+
+	// privLevel is the ipmi privilage level to used when authenicating, it is ignore
+	// when not using ipmi
+	PrivLevel string `json:"privLevel,omitempty"`
```

Comment:
Comments from downstream PR (now closed)

dhellmann 2 hours ago Member
I think we've so far been able to avoid driver-specific API fields, which helped avoid passing unused values to the driver factory and the internal API changes that triggers.

Can this be encoded into the BMC URL somehow?

 
@derekhiggins derekhiggins 1 hour ago • 
Yes, I believe it could, it was one of my earlier ideas but I instead implemented it in a similar way to "disableCertificateVerification" is there was a precedent. If encoding in the BMC url is preferred I'll push something up.

 
@dhellmann dhellmann 1 hour ago Member
disableCertificateVerifcation could potentially be used by any BMC driver (assuming it's not already), but IIUC this field only applies to IPMI, so they're not quite the same.

I would prefer using a URL parameter, but we should get input from @zaneb and others in the community (probably posting to the metal3-io version of this repo instead of the openshift version would help with that).

@derekhiggins derekhiggins 1 hour ago Author Member
Ack, thanks, will close in favour of discussing on
metal3-io#853

Reply:
I agree with Doug that we don't want this as a separate field in the CRD.

Do we really need this to be settable on a per-host basis? Couldn't it be in an environment variable or something?

Reply:
> I agree with Doug that we don't want this as a separate field in the CRD.
ok, will push up an alternative

> Do we really need this to be settable on a per-host basis? Couldn't it be in an environment variable or something?
In ironic it is set on a per-host basis, we could have BMO set it per host based on a global setting but some day somebody would need it to be per-host.